File: app/src/main/java/com/zygne/zygnearch/domain/presentation/ui/MainActivity.java
package com.zygne.zygnearch.domain.presentation.ui;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

import com.zygne.zygnearch.R;
import com.zygne.zygnearch.domain.presentation.presenters.concrete.MainPresenterImpl;
import com.zygne.zygnearch.domain.presentation.presenters.virtual.MainPresenter;
import com.zygnearchitecture.domain.executor.implementation.ThreadExecutor;
import com.zygnearchitecture.threads.AndroidThread;

public class MainActivity extends AppCompatActivity implements
        MainPresenter.View {

    private MainPresenter presenter;

    private TextView tvMain;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        tvMain = findViewById(R.id.tv_main);

        presenter = new MainPresenterImpl(ThreadExecutor.getInstance(),
                AndroidThread.getInstance(),
                this);

        presenter.start();
    }

    @Override
    public void onMainCompleted() {

        tvMain.setText("Presenter has finished");
    }

    @Override
    public void showProgress() {
        tvMain.setText("Presenter has started");
    }

    @Override
    public void hideProgress() {

    }
}

File: app/src/main/java/com/zygne/zygnearch/domain/presentation/ui/MainActivity.kt
package com.zygne.zygnearch.domain.presentation.ui

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.TextView
import com.zygne.zygnearch.R
import com.zygne.zygnearch.domain.presentation.presenters.concrete.MainPresenterImpl
import com.zygne.zygnearch.domain.presentation.presenters.virtual.MainPresenter
import com.zygnearchitecture.domain.executor.implementation.ThreadExecutor
import com.zygnearchitecture.threads.AndroidThread

class MainActivity : AppCompatActivity(), MainPresenter.View {
    private var presenter: MainPresenter? = null

    private var tvMain: TextView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        tvMain = findViewById(R.id.tv_main)
        presenter = MainPresenterImpl(ThreadExecutor.instance,
                AndroidThread.instance,
                this)
        presenter?.start()
    }

    override fun onMainCompleted() {
        tvMain!!.text = "Presenter has finished"
    }

    override fun showProgress() {
        tvMain!!.text = "Presenter has started"
    }

    override fun hideProgress() {}
}